Automattic/jetpack
# Sync: add a new method, do_only_first_initial_sync

#### Changes proposed in this Pull Request:

* Add a new method, `do_only_first_initial_sync`, which is used to trigger an initial full sync only when one hasn't already been performed. 
* This method will be useful for plugins that want to run an initial full sync on sites that haven't already had an initial sync.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Install and activate this branch.
2. Connect to WPCOM.
3. Check that the full sync status option is populated with data. The `started` and `finished` values should be  timestamps. `wp option get jetpack_sync_full_status`
4. Using `wp shell`, try to start an initial sync as shown below. This should return false because an initial sync has already been run.
`return Automattic\Jetpack\Sync\Actions::do_only_first_initial_sync();`
5. Clear the `jetpack_sync_full_status` option: `wp option delete jetpack_sync_full_status`
6. Using `wp shell`, try to start an initial sync. This should return `null` because the full sync was started.
`return Automattic\Jetpack\Sync\Actions::do_only_first_initial_sync();`
